x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -531,6 +533,166 @@ class AgentPolicyService {
+            'Agent policy requires Elastic Defend integration to set tamper protection to true'
+          );
+          // force agent policy to be false if elastic defend is not present
+          agentPolicy.is_protected = false;
+        }
+
+        if (existingAgentPolicy.is_managed && !options?.force) {
+          Object.entries(agentPolicy)
+            .filter(([key]) => !KEY_EDITABLE_FOR_MANAGED_POLICIES.includes(key))
+            .forEach(([key, val]) => {
+              if (!isEqual(existingAgentPolicy[key as keyof AgentPolicy], val)) {
+                throw new HostedAgentPolicyRestrictionRelatedError(`Cannot update ${key}`);
+              }
+            });
+        }
+
+        const { monitoring_enabled: monitoringEnabled } = agentPolicy;
+        const packagesToInstall = [];
+        if (!existingAgentPolicy.monitoring_enabled && monitoringEnabled?.length) {
+          packagesToInstall.push(FLEET_ELASTIC_AGENT_PACKAGE);
+        }
+        if (packagesToInstall.length > 0) {
+          await bulkInstallPackages({
+            savedObjectsClient: soClient,
+            esClient,
+            packagesToInstall,
+            spaceId: options?.spaceId || DEFAULT_SPACE_ID,
+            authorizationHeader: options?.authorizationHeader,
+          });
+        }
+        if (
+          existingAgentPolicy.status === agentPolicyStatuses.Inactive &&
+          agentPolicy.status !== agentPolicyStatuses.Active
+        ) {
+          throw new FleetError(
+            `Agent policy ${agentPolicy.id} cannot be updated because it is ${existingAgentPolicy.status}`
+          );
+        }
+
+        if (options.removeProtection) {
+          logger.info(`Setting tamper protection for Agent Policy ${agentPolicy.id} to false`);
+        }
+
+        if (!options.skipValidation) {
+          await validateOutputForPolicy(
+            soClient,
+            agentPolicy,
+            existingAgentPolicy,
+            getAllowedOutputTypeForPolicy(existingAgentPolicy)
+          );
+        }
+
+        policiesToUpdate.push(agentPolicy);
+      } catch (error) {
+        failedPolicies.push({ agentPolicy, error });
+        return;
+      }
+    });
+
+    const { saved_objects: updatedPolicies } = await soClient.bulkUpdate<AgentPolicySOAttributes>(
+      policiesToUpdate.map((agentPolicy) => {
+        const { id: _id, agents: _agents, ...agentPolicyWithoutImmutableFields } = agentPolicy;
+        return {
+          type: SAVED_OBJECT_TYPE,
+          id: agentPolicy.id,
+          attributes: {
+            ...agentPolicyWithoutImmutableFields,
+            ...(options.bumpRevision
+              ? { revision: existingAgentPolicies.items[0].revision + 1 }
+              : {}),
+            ...(options.removeProtection
+              ? { is_protected: false }
+              : { is_protected: agentPolicy.is_protected }),
+            updated_at: new Date().toISOString(),
+            updated_by: options?.user?.username || 'system',
```

Comment:
For the most part, I see username's defaulting to `elastic` and not `system` - but I'll defer to Fleet team on that.